app/actions/AuthenticatedActionRefiner.scala

```diff
@@ -68,24 +67,40 @@ class AuthenticatedActionRefiner @Inject() (
       }
   }
 
-  private def redirectToLoginPage(request: Request[_]): Result = {
-    val isStartEndpoint = request.uri.endsWith("/start")
-
-    val returnUrl =
-      if (isStartEndpoint) {
-        request.uri
-      } else if (appConfig.vatEnabled) {
-        routes.WhichTaxRegimeController.whichTaxRegime.url
-      } else {
-        routes.LandingController.epayeLandingPage.url
-      }
-
+  private def redirectToLoginPage(request: Request[_]): Result =
     Redirect(
       appConfig.BaseUrl.gg,
-      Map("continue" -> Seq(appConfig.BaseUrl.essttpFrontend + returnUrl), "origin" -> Seq("essttp-frontend"))
+      Map("continue" -> Seq(appConfig.BaseUrl.essttpFrontend + loginContinueToUrl(request)), "origin" -> Seq("essttp-frontend"))
     )
-  }
 
   override protected def executionContext: ExecutionContext = cc.executionContext
 
 }
+
+@Singleton
+class ContinueToLandingPagesAuthenticatedActionRefiner @Inject() (
```

Comment:
I couldn't think of a better name for this or the one below @barnesjake - open to suggestions!

Reply:
lol tbh it does what it says on the tin, no objections here